dogfood.md

```diff
@@ -0,0 +1,167 @@
+
+### iOS
+
+The iOS SDK has all of the features we need, and iOS has an Apollo GraphQL library as well.  Like Cordova and Android there is a lot of experience we can draw on for developing the app.
+
+
+## Data Model and API layer
+
+The following is a graphql schema for the proposed data model.  The GraphQL schema also includes the public APIs that will be implemented.
+
+### Sample Schema
+
+```graphql
+type Profile {
+  id: ID! @unique
+  email: String!
+  display_name: String!
+  biography: String!
+  pictureUrl: String!
+}
+
+type MemeList {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  memes: [Meme!]!
+}
+
+type Meme {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  photoUrl: String!
+  comments: [Comment!]!
+}
+
+type Comment {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  comment: String!
+}
+
+type Query {
+  profile(email: String!):Profile!
+  usersMemes(email: String!): [Meme!]!
+  feed(memeListId: ID!): [Meme!]!
+  meme(memeId: ID!):Meme
+}
+
+type Mutation {
+  createProfile(email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  updateProfile(id: ID!, email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  deleteProfile(id: ID!):Profile!
+
+  postMeme(picture: [Int!]!, topComment: String!,bottomComment: String!, memeListId: ID!): Meme!
+  deleteMeme(id: ID!): Meme!
+  likeMeme(id: ID!): Meme!
+  unlikeMeme(id: ID!): Meme!
+
+  postComment(memeId: ID!, comment: String!): Meme!
+  deleteComment(id: ID!): Meme!
+}
+```
+
+## Mobile-Core features used
+
+Memeolist will be build using technologies provided with the mobile-core service catalog.  Additional services will be created as necessary using RHOAR runtimes and exposed via a custom connector.  Finally, we will use this as an opportunity to test graphql style APIs which will serve as the basis for the future sync project.
+
+### Custom Service Connector
+
+The final meme images will be generated server side using a RHOAR runtime (vert.x, node.js, or Wildfly Swarm) and exposed using the custom service connector feature.  This means that our SDKs will be responsible for upload the image, the author metadata, and the message to be displayed on it and the RHOAR service will compose that data into a Meme post and save it to a persistent data store.
+
+### Keycloak
+
+Social integration will be provided with KeyCloak and we will use our auth SDKs.  We should also provide a templated realm and a script so that community users can plug in their developer credentials if they wish to host their own version of memeolist.
+
+### Push
+
+We should use UPS to notify users of activity on memes they have published or liked.
+
+### Metrics
+
+Our metrics SDK should be used to collect the following information 
```

Comment:
cc @wei-lee as these use cases may help when writing the metrics phase 2 proposal for custom metrics